client/public/js/frame.js

```diff
@@ -1,41 +1,29 @@
 (function() {
-  function evalSources(sources) {
-    var modules = {};
-    function dirname(str) {
-      var ix = str.lastIndexOf("/");
-      return ix < 0 ? "" : str.slice(0, ix);
-    }
-    function resolvePath(a, b) {
-      if (b[0] === "." && b[1] === "/") {
-        return dirname(a) + b.slice(1);
-      }
-      if (b[0] === "." && b[1] === "." && b[2] === "/") {
-        return dirname(dirname(a)) + b.slice(2);
-      }
-      return b;
-    }
-    return function load(name) {
-      if (modules[name]) {
-        return modules[name].exports;
-      }
-      function require(path) {
-        return load(resolvePath(name, path));
-      }
-      var module = modules[name] = { exports: {} };
-      new Function("module", "exports", "require", sources[name])(module, module.exports, require);
-      return module.exports;
-    };
-  }
-
   var parent;
 
+  // Ensures `main` is only called once.
+  this.window.runCount = 0;
+
   document.addEventListener("DOMContentLoaded", function() {
     window.addEventListener("message", function(event) {
       parent = event.source;
       parent.postMessage("trypurescript", "*");
-      var file = evalSources(event.data)("<file>");
-      if (file.main && typeof file.main === "function") {
-        file.main();
+      const code = `
+      ${event.data.code}
+      if (window.runCount === 0) {
+        main();
+        // bump runCount to ensure we don't re-run this.
+        window.runCount = 1;
+      }`;
+      const scriptEl = document.createElement("script");
+      scriptEl.type = "module";
+      // See https://stackoverflow.com/a/6433770
+      try {
+        scriptEl.appendChild(document.createTextNode(code));
+      } catch (e) {
+        scriptEl.text = code;
```

Comment:
Do you have a specific browser that you are guarding against with this code? I would be surprised if any of the browsers that support ES modules don't support the first method.

Reply:
> Do you have a specific browser that you are guarding against with this code? 

No, I don't. I'll update this code.